src/calculation/DetailedResultsBuilder.ts

```diff
@@ -52,7 +52,9 @@ export function createPopulationValues(
         episodeResult.populationResults.forEach(popResult => {
           const measureObsPop = populationResults.find(
             pop =>
-              pop.populationType === PopulationType.OBSERV && popResult.criteriaExpression === pop.criteriaExpression
+              pop.populationType === PopulationType.OBSERV &&
+              popResult.criteriaExpression === pop.criteriaExpression &&
+              pop.criteriaReferenceId === popResult.criteriaReferenceId
```

Comment:
I remember when the only measure type where we were dealing with measure observations was a CV episode measure (cms111). This measure had one measure observation without a criteriaReference extension, and so we assumed it was observing the measure observation.

This was a while ago though, and looking at the IG, it looks like even if there's only 1 measure observation in a CV measure, it SHALL use the criteriaReferenceId: https://build.fhir.org/ig/HL7/cqf-measures/measure-conformance.html#continuous-variable-measure

I was initially concerned that we could be dealing with measure observations without a criteriaReference extension, which I think this code would break on and not find the result. But just confirming that maybe we don't have to worry about that anymore?

If we wanted to maintain backwards compatibility with that assumption, we'd want to only do the criteriaReferenceId check if it exists on the measure observation population def to begin with.

Although now that I typed this out, it might technically still work for `undefined === undefined` lol, but i still think we should talk about it